### PR TITLE
Gas Estimator: Add unit tests, change getPrice to _getPrice

### LINKED ISSUE #1183: Add tests for GasEstimator
Right now the GasEstimator has no tests. This critical infrastructure needs to be tested

### PR DESCRIPTION
**Pull request by nicholaspai**
Resolves #1183 

`getPrice()` was only called internally by `_update()`, so I re-named to `_getPrice()` to better highlight that a client should be calling `getCurrentFastPrice()`.

Signed-off-by: Nick Pai <npai.nyc@gmail.com>

### CODE REVIEW

File: financial-templates-lib/test/GasEstimator.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    it("Returns gas prices in wei before initial update", () => {
+        assert.equal(gasEstimator.defaultFastPriceGwei, gasEstimator.getCurrentFastPrice()/1e9);
+    });
+    it("Returns gas prices in wei after update", async () => {
+        await gasEstimator.update();
+        const latestFastGasPrice = gasEstimator.getCurrentFastPrice() / 1e9;
+        if (latestFastGasPrice === gasEstimator.defaultFastPriceGwei) {
+            console.log(`API Request to ethgasstation.info failed, using default gas price in Gwei: ${latestFastGasPrice}`);
+        } 
+        assert(latestFastGasPrice > 0);
```
Comment (chrismaree) [inline financial-templates-lib/test/GasEstimator.js:28]: Do you want to call the API directly from the test script and validate the output? might be redundant though but this test does not actually check the price changes from the `defaultFastPriceGwei`.

Not sure, it's hard to test due to the coupling with `ethGasStation`.

Comment (nicholaspai) [reply financial-templates-lib/test/GasEstimator.js:28]: So I thought about that but what if the API returns the same result as `defaultFastPriceGwei`?

File: financial-templates-lib/GasEstimator.js [hunk 2]
```diff
@@ -42,17 +42,17 @@ class GasEstimator {
...
-    this.lastFastPriceGwei = returnedPrice;
-  };
-
   // Returns the current fast gas price in Wei, converted from the stored Gwei value.
   getCurrentFastPrice = () => {
     return this.lastFastPriceGwei * 1e9;
   };
 
-  getPrice = async url => {
+  _update = async () => {
```
Comment (mrice32) [inline financial-templates-lib/GasEstimator.js:50]: Do you think we should replicate the `update` and `forceUpdate` setup/terminology we use in other clients?

File: financial-templates-lib/GasEstimator.js [hunk 3]
```diff
@@ -42,17 +42,17 @@ class GasEstimator {
...
+    let returnedPrice = await this._getPrice(url);
+    this.lastFastPriceGwei = returnedPrice;
```
Comment (mrice32) [inline financial-templates-lib/GasEstimator.js:52]: nit: why not just one-line this:
```
this.lastFastPriceGwei = await this._getPrice(url);
```

File: financial-templates-lib/GasEstimator.js (hunk 2 above)
Comment (nicholaspai) [reply financial-templates-lib/GasEstimator.js:50]: I think we can add `forceUpdate` if we see a need to do it, but none of the clients knowingly trigger a gasPrice change, unlike how they knowingly trigger an EMP Client state change. 

I think a situation that could potentially call for it is if the client bots detect heightened ETH network fee volatility, they might force an update.

File: financial-templates-lib/GasEstimator.js (hunk 3 above)
Comment (nicholaspai) [reply financial-templates-lib/GasEstimator.js:52]: good catch

Comment (nicholaspai) [reply financial-templates-lib/GasEstimator.js]: done

File: financial-templates-lib/GasEstimator.js (hunk 2 above)
Comment (mrice32) [reply financial-templates-lib/GasEstimator.js:50]: Okay, SGTM. We can always add later if we find we need it.

**Pull request merged**